addons/hr/models/res_users.py

```diff
@@ -1,21 +1,156 @@
+    visa_expire = fields.Date(related='employee_id.visa_expire', readonly=False, related_sudo=False)
+    google_drive_link = fields.Char(related='employee_id.google_drive_link', readonly=False, related_sudo=False)
+    additional_note = fields.Text(related='employee_id.additional_note', readonly=False, related_sudo=False)
+
+
+    def __init__(self, pool, cr):
+        """ Override of __init__ to add access rights.
+            Access rights are disabled by default, but allowed
+            on some specific fields defined in self.SELF_{READ/WRITE}ABLE_FIELDS.
+        """
+        hr_readable_fields = [
+            'employee_ids',
+            'employee_id',
+            'active',
+            'parent_id',
+            'child_ids',
+        ]
+
+        hr_writable_fields = [
+            'display_name',
+            'job_title',
+            'work_phone',
+            'phone',
+            'mobile_phone',
+            'work_email',
+            'category_ids',
+            'department_id',
+            'address_id',
+            'work_location',
+            'parent_id',
+            'coach_id',
+            'notes',
+            'address_home_id',
+            'is_address_home_a_company',
+            'km_home_work',
+            'employee_bank_account_id',
+            'employee_country_id',
+            'identification_id',
+            'passport_id',
+            'gender',
+            'birthday',
+            'place_of_birth',
+            'country_of_birth',
+            'marital',
+            'spouse_complete_name',
+            'spouse_birthdate',
+            'children',
+            'emergency_contact',
+            'emergency_phone',
+            'visa_no',
+            'permit_no',
+            'visa_expire',
+            'google_drive_link',
+            'additional_note',
+        ]
+
+        init_res = super(User, self).__init__(pool, cr)
+        # duplicate list to avoid modifying the original reference
+        type(self).SELF_READABLE_FIELDS = type(self).SELF_READABLE_FIELDS + hr_readable_fields + hr_writable_fields
+        type(self).SELF_WRITEABLE_FIELDS = type(self).SELF_WRITEABLE_FIELDS + hr_writable_fields
+        return init_res
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        # When the front-end loads the views it gets the list of available fields
+        # for the user (according to its access rights). Later, when the front-end wants to
+        # populate the view with data, it only asks to read those available fields.
+        # However, in this case, we want the user to be able to read/write its own data,
+        # even if they are protected by groups.
+        # We make the front-end aware of those fields by sending all field definitions.
+        return super(User, self.sudo()).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
```

Comment:
And another separatist group. ^^

This breaks the security model. As a portal user, I could be able to access to the res.users form content, the sniffers are quite happy ;)

At least do something like 
``` python
if not self.env.user.share:
    self = self.sudo()
return super(User, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
```

Reply:
I guess the feature is breaking the security model by design :/ 
I'll apply your comment